modules/metastorage/src/main/java/org/apache/ignite/internal/metastorage/server/ReadOperationForCompactionTracker.java

```diff
@@ -36,17 +36,23 @@
  * <p>Expected usage:</p>
  * <ul>
  *     <li>Before starting execution, the reading command invoke {@link #track} with its ID and the compaction revision that is currently
- *     set ({@link MetaStorageCompactionManager#setCompactionRevisionLocally}/{@link KeyValueStorage#setCompactionRevision}).</li>
+ *     set ({@link KeyValueStorage#setCompactionRevision}).</li>
  *     <li>After completion, the reading command will invoke {@link #untrack} with the same arguments as when calling {@link #track},
  *     regardless of whether the operation was successful or not.</li>
  *     <li>{@link #collect} will be invoked only after a new compaction revision has been set
- *     ({@link MetaStorageCompactionManager#setCompactionRevisionLocally}/{@link KeyValueStorage#setCompactionRevision}) for a new
- *     compaction revision.</li>
+ *     ({@link KeyValueStorage#setCompactionRevision}) for a new compaction revision.</li>
  * </ul>
  */
 public class ReadOperationForCompactionTracker {
     private final Map<ReadOperationKey, CompletableFuture<Void>> readOperationFutureByKey = new ConcurrentHashMap<>();
 
+    private final AtomicLong longOperationIdGenerator = new AtomicLong();
+
+    /** Generates the next read operation ID. Thread-safe. */
+    public long generateLongReadOperationId() {
```

Comment:
what do you mean by "Long read operation ID"? Is it "long" because of the return type or because the read operation takes a long time?

Reply:
The return type is "long", can you suggest a better name?

Reply:
yes, I think straightforward `generateReadOperationId` is better

Reply:
Fix it